### ISSUE TITLE
Paginate the annotation table on annotation shape type

### ISSUE THREAD

**Issue by codemonkey800** [chanzuckerberg/cryoet-data-portal#581]
If the files within an annotation have different shape types, they should be rendered on separate rows.

We'll need to handle pagination a bit carefully. For any given page, the table should have at most 20 annotations on the table. Assume we query the backend and get 20 annotations. If all 20 have files with the same shape type, then the table will render correctly

However, if the first 7 annotations has all 3 files each with different shape types, then we'll only be able to render 20 out of the 21 files using the existing pagination logic. Additionally, the remaining the 13 annotations may never get rendered because when the user clicks on the next page, it paginates only on the `annotations` query and not the `files` query

One approach is to query all of the annotation and file data and handle pagination on the frontend server, but does not feel very ideal because we need to query every single annotation and file for a run which may never get visited if the user doesn't go through each page

**Comment by junxini** [comment]
note to self: added priority P1 only because the use case of having multiple shape types per annotation doesn't currently exist. however, this should be on the higher end of the P1 list so we can implement before the first use case appears.

**Comment by manasaV3** [comment]
I agree that there are better ways to go for this than loading all the data in the front end. As we don't have too many annotations, the pagination will not be an issue for now. 
Hopefully, this helps with the long-term solution: https://github.com/chanzuckerberg/cryoet-data-portal/issues/581

**Comment by manasaV3** [comment]
This work is now unblocked by this interim solution: https://github.com/chanzuckerberg/cryoet-data-portal/issues/582#issuecomment-2251260752

**Comment by bchu1** [comment]
This query would actually allow us to paginate unique `annotation` and `shape_type` pairs properly:
```
query MyQuery {
  annotation_files(
    where: {annotation: {tomogram_voxel_spacing: {run_id: {_eq: 121}}}}
    distinct_on: [annotation_id, shape_type]
    limit: 10
  ) {
    annotation_id
    annotation {
      ground_truth_status
    }
  }
}
```

**Comment by bchu1** [comment]
@manasaV3 @codemonkey800 Sad news: Looks like the above query won't work because Postgres requires that `DISTINCT ON` columns must be a subset of the `ORDER BY` columns. We're ordering by `ground_truth_status` so Hasura errors.

Theoretically if Hasura supported `distinct_on` on nested fields we could make this work by adding `ground_truth_status` into the `distinct_on` but it only supports the fields of the immediate object.

I'll unassign myself for now until the new schema comes along.

**Comment by bchu1** [comment]
Example query that I thought would work:
```
query MyQuery {
  annotation_files(
    distinct_on: [annotation_id, shape_type]
    order_by: {annotation: {ground_truth_status: asc}}
  ) {
    annotation {
      id
    }
  }
}
```

Hasura response:
```
{
  "errors": [
    {
      "extensions": {
        "code": "validation-failed",
        "path": "$.selectionSet.annotation_files.args"
      },
      "message": "\"distinct_on\" columns must match initial \"order_by\" columns"
    }
  ]
}
```

**Comment by bchu1** [comment]
Hasura documentation on limitation: https://hasura.io/docs/latest/queries/postgres/distinct-queries/

Postgres documentation on limitation: https://www.postgresql.org/docs/current/sql-select.html#:~:text=chosen%20query%20plan.-,DISTINCT%20Clause,-If%20SELECT%20DISTINCT

**Comment by bchu1** [comment]
Reassigning myself, Manasa reminded me of the hack we can use temporarily (filter out .zarr files).

**Comment by bchu1** [comment]
@manasaV3 @junxini Here's the ticket for the final query https://github.com/chanzuckerberg/cryoet-data-portal/issues/581
